File: python/clock/clock.py
class Clock:
    def __init__(self, hour, minute):
        self.hour = hour
        self.minute = minute
        self.set_time()

    def __str__(self):
        return '{0:02d}:{1:02d}'.format(self.hour, self.minute)

    def __eq__(self, clock):
        return str(self) == str(clock)

    def add(self, minutes):
        self.minute += minutes
        return self.set_time()

    def set_time(self):
        self.hour += self.minute // 60
        self.hour %= 24
        self.minute %= 60
        return self

File: elixir/clock/lib/clock.ex
defmodule Clock do
  defstruct hour: 0, minute: 0

  defimpl String.Chars, for: Clock do
    def padding(p) do
      String.pad_leading(Integer.to_string(p), 2, "0")
    end

    def to_string(c) do
      "#{padding(c.hour)}:#{padding(c.minute)}"
    end
  end

  @doc """
  Returns a clock that can be represented as a string:

      iex> Clock.new(8, 9) |> to_string
      "08:09"
  """
  @spec new(integer, integer) :: Clock
  def new(hour, minute) do
    add(%Clock{}, hour * 60 + minute)
  end

  @doc """
  Adds two clock times:

      iex> Clock.new(10, 0) |> Clock.add(3) |> to_string
      "10:03"
  """
  @spec add(Clock, integer) :: Clock
  def add(%Clock{hour: hour, minute: minute}, add_minute) do
    minutes = minute + add_minute
    hours = hour + Integer.floor_div(minutes, 60)
    %Clock{
      hour: Integer.mod(hours, 24),
      minute: Integer.mod(minutes, 60)
    }
  end
end
